crates/holochain_cascade/src/lib.rs

```diff
@@ -618,11 +618,7 @@ impl CascadeImpl {
         let query: GetRecordDetailsQuery =
             self.construct_query_with_data_access(action_hash.clone());
 
-        // DESIGN: we can short circuit if we have any local deletes on an action.
-        // Is this bad because we will not go back to the network until our
-        // cache is cleared. Could someone create an attack based on this fact?
```

Comment:
Nope, it's never okay to do that. Unless you're an authority, you have to go to the network for updates and deletes!

We were just short-cutting here as soon as we had any record which is obviously wrong because no new version will ever be seen.